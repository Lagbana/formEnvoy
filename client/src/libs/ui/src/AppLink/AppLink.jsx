/* Text-based link
=============================== */

import React from 'react';
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  TextStyle,
} from 'react-native';
import { Typography, Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppIcon } from '../AppIcon';

export interface AppLinkProps {
  onPress?: () => void;
  icon?: string;
  iconSize?: number;
  iconOpacity?: number;
  color?: string;
  style?: TextStyle;
}

export const AppLink: React.FC<AppLinkProps> = ({
  children,
  onPress,
  icon,
  iconSize = 32,
  iconOpacity = 0.65,
  color = 'blue',
  style,
}) => {
  const styles = StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: style?.textAlign === 'right' ? 'flex-end' : 'flex-start',
    },
    link: {
      color: Colors[color],
      fontSize: Typography.sizes.small,
      lineHeight: Typography.sizes.small * 1.2,
      fontFamily: Typography.families.bold,
      textAlign: style?.textAlign ?? 'left',
    },
    icon: {
      marginRight: 10,
      opacity: iconOpacity,
    },
  });

  return (
    <TouchableOpacity style={[styles.container, style]} onPress={onPress}>
      {icon && (
        <AppIcon
          name={icon}
          color={Colors[color]}
          size={iconSize}
          style={styles.icon}
        />
      )}
      <Text style={styles.link}>{children}</Text>
    </TouchableOpacity>
  );
};
