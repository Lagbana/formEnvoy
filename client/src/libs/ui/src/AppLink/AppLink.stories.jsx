import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppLink } from './AppLink'

storiesOf('AppLink', module)
  .add('Phone', () => <AppLink icon='phone' iconSize={32} color='blue'children={'Hello Button!'} />)
  .add('User', () => <AppLink icon='user' iconSize={32} color='blue' children={'Hello Button!'}/>)
  .add('Activity', () => <AppLink icon='activity' iconSize={32} color='blue' children={'Hello Button!'}/>)
  .add('Activity with 50% Opacity', () => (
    <AppLink icon='activity' iconSize={32} color='blue' iconOpacity={50} children={'Hello Button!'}/>
  ))
