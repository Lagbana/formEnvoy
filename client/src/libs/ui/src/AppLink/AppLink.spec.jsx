import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppLink } from './AppLink'

configure({ adapter: new Adapter() })

test('<AppLink /> renders correctly', () => {
  const component = shallow(<AppLink />)
  component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})
