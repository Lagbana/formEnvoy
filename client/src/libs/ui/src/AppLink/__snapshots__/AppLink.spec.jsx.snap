// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLink /> renders correctly 1`] = `""`;
